br/pkg/encryption/master_key/multi_master_key_backend.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 PingCAP, Inc. Licensed under Apache-2.0.
+
+package encryption
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/encryptionpb"
+)
+
+const (
+	defaultBackendCapacity = 5
+)
+
+type MultiMasterKeyBackend struct {
+	backends []Backend
+}
+
+func NewMultiMasterKeyBackend(masterKeysProto []*encryptionpb.MasterKey) (*MultiMasterKeyBackend, error) {
+	if masterKeysProto == nil && len(masterKeysProto) == 0 {
+		return nil, errors.New("must provide at least one master key")
+	}
+	var backends = make([]Backend, 0, defaultBackendCapacity)
+	for _, masterKeyProto := range masterKeysProto {
+		backend, err := CreateBackend(masterKeyProto)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		backends = append(backends, backend)
+	}
+	return &MultiMasterKeyBackend{
+		backends: backends,
+	}, nil
+}
+
+func (m *MultiMasterKeyBackend) Decrypt(ctx context.Context, encryptedContent *encryptionpb.EncryptedContent) ([]byte, error) {
+	if len(m.backends) == 0 {
+		return nil, errors.New("internal error: should always contain at least one backend")
+	}
+
+	var errMsgs = make([]string, 0, defaultBackendCapacity)
+	for _, masterKeyBackend := range m.backends {
```

Comment:
We can use [`multierr`](https://pkg.go.dev/go.uber.org/multierr) here.

Reply:
oh that's handy, thanks!